src/yall.mjs

```diff
@@ -58,6 +59,7 @@ export default function (options) {
       if (dataAttrs[dataAttrIndex] in element.dataset) {
         win["requestAnimationFrame"](() => {
           element.setAttribute(dataAttrs[dataAttrIndex], element.dataset[dataAttrs[dataAttrIndex]]);
+          element.classList.remove(lazyClass);
```

Comment:
I moved this oppertation here so that the class gets removed regardless of weather or not intersection observer is being used.